### DISCUSSION TITLE
[notice message]  How do I display the contents of a notification message in a new buffer

### DISCUSSION THREAD

**Discussion by iamxiaojianzheng** [LazyVim/LazyVim#3704] [Q&A] [answered]
For example, if I type `digraphs` in command mode, this will be displayed in the notification message. 

But I want to show it in a new buffer.

**Comment by dpetka2001** [comment]
https://superuser.com/questions/1257942/how-can-i-get-a-navigable-editable-buffer-with-vims-messages

**Reply by dpetka2001** [reply, marked as answer]
I tried the `put` example and it worked. Probably the one which redirects output to a file should work as well. If you try the `put` method you might want to open a new buffer (`<leader>fn`) to do it, otherwise it will paste the contents in your current buffer.

**Reply by iamxiaojianzheng** [reply]
I just watched a video on a vim tutorial where the execution `:digraphs` example shows the results directly in a buffer. 

I'm not sure that's the difference between vim and neovim? Or is it because of lazyvim's configuration?

**Reply by dpetka2001** [reply]
It's not an editable buffer. It's just a temporary buffer. In LazyVim you can do `:Noice all` and then you can maximize that window. But again that's not modifiable. I don't know if you need a modifiable buffer, but the `put` method puts the contents in a modifiable buffer and you can do whatever you want there.

**Reply by folke** [reply]
Actually, there's a noice keymap in LazyVim for exactly this :)

Just do `:ls` and instead of pressing `<CR>`, use `<S-CR>`. Enjoy :)

**Reply by iamxiaojianzheng** [reply]
This seems to be a bug, and  `:ls <S-CR>` is different from the result of `lua require("noice").redirect("ls")`.

**Reply by iamxiaojianzheng** [reply]
@folke @dpetka2001 

Does `<S-Enter>` work on your terminal ?

This mapping does not seem to be working properly.

```lua
-- <S-Enter> or <S-CR> I've tried them all
{ "<S-Enter>", function() require("noice").redirect(vim.fn.getcmdline()) end, mode = "c", desc = "Redirect Cmdline" },
```

I modified it to `C-a`, it can work normally.
```lua
{ "<C-a>", function() require("noice").redirect(vim.fn.getcmdline()) end, mode = "c", desc = "Redirect Cmdline" },
```

**Reply by folke** [reply]
Works correctly on wezterm and kitty for me

**Reply by iamxiaojianzheng** [reply]
Got it. This may be a terminal problem.

I'm using Windows terminal.

**Reply by folke** [reply]
I highly recommend to use wezterm on Windows.

**Reply by dpetka2001** [reply]
Enter `insert` mode and press `<C-v><S-enter>`. What characters are printed out? That's what your terminal interprets the key sequence when pressed.

**Reply by iamxiaojianzheng** [reply]
> Enter `insert` mode and press `<C-v><S-enter>`. What characters are printed out? That's what your terminal interprets the key sequence when pressed. 

The result is `^@`.

**Reply by dpetka2001** [reply]
It's a terminal issue as to how it interprets the key sequence. Instead of `<C-a>` change that to `<C-@>` and you should be able to use `<S-enter>` normally.
